pandas/core/frame.py

```diff
@@ -8923,29 +8923,29 @@ def update(
 
         other = other.reindex(self.index)
 
-        for col in self.columns.intersection(other.columns):
-            this = self[col]._values
-            that = other[col]._values
+        cols = self.columns.intersection(other.columns)
```

Comment:
Calling ``._values`` on multiple columns converts to a common dtype for all columns, probably object as soon as you have non-numeric columns. This is terrible from a performance perspective, so doing this at once is actually a lot slower in real world use cases than doing one column at a time

Reply:
Ah, understood.  Thank you for taking the time to explain.